waiter/src/waiter/request_log.clj

```diff
@@ -45,25 +46,29 @@
       content-length (assoc :request-content-length content-length)
       content-type (assoc :request-content-type content-type)
       request-time (assoc :request-time (du/date-to-str request-time))
+      server-port (assoc :server-port server-port)
       user-agent (assoc :user-agent user-agent))))
 
 (defn response->context
   "Convert a response into a context suitable for logging."
   [{:keys [authorization/method authorization/principal backend-response-latency-ns descriptor latest-service-id
            get-instance-latency-ns handle-request-latency-ns headers instance instance-proto protocol status
            waiter-api-call?] :as response}]
-  (let [{:keys [service-id service-description]} descriptor
-        {:strs [content-length content-type grpc-status server]} headers
+  (let [{:keys [service-id service-description source-tokens]} descriptor
+        token (some->> source-tokens (map #(get % "token")) seq (str/join ","))
```

Comment:
Is this sorted in some deterministic way? Is it worth doing a sort here to make sure partial tokens always show up in the same order? That seems important for filtering logs based on this value.

Reply:
`source-tokens` are sorted in the order the user specified them as that also affects the service description computed in the end. So, it would produce them is the same order as they are specified in the `x-waiter-token` header.

Reply:
Ah, that makes sense. I was thinking of disjoint tokens, but they can overlap too so the merge order matters. Thanks.